src/track/track.cpp

```diff
@@ -1109,6 +1121,37 @@ void Track::importPendingCueInfosMarkDirtyAndUnlock(
     emit cuesUpdated();
 }
 
+QMap<int, MacroPointer> Track::getMacros() const {
+    QMutexLocker lock(&m_qMutex);
+    return m_macros;
+}
+
+void Track::setMacros(QMap<int, MacroPointer> macros) {
+    QMutexLocker lock(&m_qMutex);
+    m_macros = macros;
+}
+
+void Track::addMacro(int slot, MacroPointer macro) {
+    QMutexLocker lock(&m_qMutex);
+    m_macros.insert(slot, macro);
+    if (macro->isDirty()) {
+        markDirtyAndUnlock(&lock);
+    }
+}
+
+bool Track::isDirty() {
+    QMutexLocker lock(&m_qMutex);
+    if (m_bDirty) {
+        return true;
+    }
+    for (MacroPointer macro : m_macros) {
+        if (macro->isDirty()) {
```

Comment:
```suggestion
    for (const MacroPointer& pMacro : qAsConst(m_macros)) {
        if (pMacro->isDirty()) {
```

Reply:
that is nonsense, why would you create a const reference to a pointer?

Reply:
The pointer should be immutable to prevent modification, i.e. unintended assignment. Some languages enforce this as the default, C++ does not.

Using a const-reference also is an optimization that prevents unneeded, internal book keeping caused by copying the smart pointer. Remember, there is an atomic involved, which requires a memory barrier, i.e. cache synchronization between CPU cores.

Please be careful by rejecting a recommendation as "nonsense" without being aware of some subtle details.

Reply:
Sorry, it really confused me. That does seem to make sense.

Reply:
what's the point of `qAsConst` though?

Reply:
https://github.com/KDE/clazy/blob/master/docs/checks/README-range-loop.md